frontend/controller/actions/group.js

```diff
@@ -914,41 +912,92 @@ export default (sbp('sbp/selectors/register', {
-    const state = await sbp('chelonia/contract/state', params.contractID)
-    const data = params.data
+    const { contractID, data } = params
+    const state = await sbp('chelonia/contract/state', contractID)
     const proposalHash = data.proposalHash
     const proposal = state.proposals[proposalHash]
     const type = proposal.data.proposalType
-    let passPayload
-
-    if (data.vote === VOTE_FOR) {
-      passPayload = {}
-      if (oneVoteToPass(state, proposalHash)) {
-        if (type === PROPOSAL_INVITE_MEMBER) {
-          passPayload = await createInvite({
-            contractID: params.contractID,
-            invitee: proposal.data.proposalData.memberName,
-            creatorID: proposal.creatorID,
-            expires: state.settings.inviteExpiryProposal
-          })
-        }
+
+    const willBePassed = oneVoteToPass(state, proposalHash)
+    const willBeFailed = oneVoteToFail(state, proposalHash)
+    const isVoteFor = data.vote === VOTE_FOR
+    const isVoteAgainst = !isVoteFor
+    let passPayload = isVoteFor ? {} : undefined
+    let proposalToSend
+
+    if (willBePassed && isVoteFor) {
+      if (type === PROPOSAL_INVITE_MEMBER) {
+        passPayload = await createInvite({
+          contractID,
+          invitee: proposal.data.proposalData.memberName,
+          creatorID: proposal.creatorID,
+          expires: state.settings.inviteExpiryProposal
+        })
       }
+
+      proposalToSend = { ...extractProposalData(proposal, { proposalId: proposalHash }), status: STATUS_PASSED }
+    } else if (willBeFailed && isVoteAgainst) {
+      proposalToSend = { ...extractProposalData(proposal, { proposalId: proposalHash }), status: STATUS_FAILED }
     }
 
     return sendMessage({
       ...params,
-      data: {
-        ...data,
-        passPayload
+      data: { ...data, passPayload },
+      hooks: {
+        ...params?.hooks,
+        postpublish: (...args) => {
+          if (proposalToSend) {
+            sbp('gi.actions/group/notifyProposalStateInGeneralChatRoom', {
+              groupID: contractID,
+              proposal: proposalToSend
+            })
+            return params.hooks?.postpublish?.(...args)
+          }
+        }
       }
     })
   }),
-  ...encryptedAction('gi.actions/group/proposalCancel', L('Failed to cancel proposal.')),
+  ...encryptedAction('gi.actions/group/proposalCancel', L('Failed to cancel proposal.'), async function (sendMessage, params) {
+    const { contractID, data } = params
+    const state = await sbp('chelonia/contract/state', contractID)
+    const proposal = state.proposals[data.proposalHash]
+    const proposalToSend = { ...extractProposalData(proposal, { proposalId: data.proposalHash }), status: STATUS_CANCELLED }
+
+    await sendMessage(params)
+
+    await sbp('gi.actions/group/notifyProposalStateInGeneralChatRoom', { groupID: contractID, proposal: proposalToSend })
+  }),
```

Comment:
Please be consistent. Either this should be rewritten to use `onprocessed` (or `postpublish`), or the other code should be rewritten to avoid their use.

This does seem to be the simpler approach, should I vote for this method.

Reply:
Okay, I will use `postpublish` hook then.